### DISCUSSION TITLE
Icons for UI

### DISCUSSION THREAD

**Discussion by zhuiks** [ezra-bible-app/ezra-bible-app#486] [Development]
This could be the place where we discuss Ezra Bible App icon set and its implementation for the UI.

First about implementation. Currently, we use Font Awesome as a main icon source. Icon font is uploaded in the header and inserted by assigning certain class: `<i class="fas fa-book-open>`. However, in some places (InstallAssistant, WheelNav) svg icons are used (I'm guilty of that). It makes sense to have the same implementation everywhere. The way svg used in WheelNav makes the most sense.

Some icons are just two icons next to each other: 
![Screenshot from 2021-10-19 17-19-02](https://user-images.githubusercontent.com/3584119/137998679-339db055-7670-4c17-90b3-19a4f583dc69.png)
It would be nice to redraw them as one square icon. I.e. create custom icons. I think Font Awesome is not "open" enough for that. @augustin-colesnic do you have any other suggestions about icon library?

Here are the icons that we use/need:
- [x] ![Screenshot from 2021-10-19 17-45-46](https://user-images.githubusercontent.com/3584119/138001062-45e464fd-9cec-4d1e-a18c-64c2b9b87dfb.png) Bible book - used to select book of the Bible. It can be also a pictogram to show that certain tab is in "Bible book" mode
- [x] ![Screenshot from 2021-10-19 17-45-57](https://user-images.githubusercontent.com/3584119/138001124-62c18910-466d-4181-9d43-eb5efb50f911.png) Tags - used to switch tab in "tags" mode. It can be also an icon for "Show tag list button", as well as pictogram on the tab
- [x] ![Screenshot from 2021-10-19 17-46-04](https://user-images.githubusercontent.com/3584119/138001146-3deef3ac-cf0f-4379-b13d-fc5af804d4f6.png) Search - used to switch tab in "search" mode.
- [x] ![Screenshot from 2021-10-19 17-46-12](https://user-images.githubusercontent.com/3584119/138001175-f3e39108-a7da-47bf-90fb-c466aa885295.png) Settings
- [ ] Adjust font size
- [x] ![Screenshot from 2021-10-19 17-46-33](https://user-images.githubusercontent.com/3584119/138001215-56689d47-d89b-4360-b99b-5ce9f9000456.png) Information
- [x] ![Screenshot from 2021-10-19 17-50-26](https://user-images.githubusercontent.com/3584119/138001293-2be648a3-adef-464f-8728-62041bb7e853.png) Export to text document
- [ ] Export to CSV
- [ ] ![Screenshot from 2021-10-19 17-46-21](https://user-images.githubusercontent.com/3584119/138001319-72112dd2-0110-4115-bb55-d8ccb8501831.png)  Tag statistics
- [ ] Change tags - should it be the same as "tags" icon?
- [ ] Assign last tag
- [ ] ![Screenshot from 2021-10-19 17-19-02](https://user-images.githubusercontent.com/3584119/138001351-cb2656a6-fa15-49f8-bf8c-2566c47e37fc.png) New tag
- [x] ![Screenshot from 2021-10-19 17-46-46](https://user-images.githubusercontent.com/3584119/138001388-1fb36064-36aa-4495-8366-40fb0434760a.png) Compare
- [x] ![Screenshot from 2021-10-19 17-47-15](https://user-images.githubusercontent.com/3584119/138001423-4f14156a-3459-4520-9cf6-3a3bee136d8b.png) Context
- [ ] Copy to clipboard
- [ ] Cross-references
- [ ] Add note
- [x] ![Screenshot from 2021-10-19 17-52-52](https://user-images.githubusercontent.com/3584119/138001469-35e67b5c-b055-480f-b8e9-351077a9ab38.png) Has note
- [ ] Night / Day mode
- [ ] Dictionary
- [ ] Has Strong Numbers
- [ ] Show Headings
- [ ] Footnotes

Any other icons ideas that I'm missing?

**Comment by tobias-klein** [comment]
Thank you @zhuiks! This is a good overview.

To anyone reading this: Contributing with icon work and/or graphical design would be a good way to contribute.

The font awesome icons are SVG based and that allows for good scalability. We should keep the constraint that icons just be SVG.

**Reply by 42sol-eu** [reply]
that sounds like a task that is up my alley. 

Do you know the remix icon project? It might be a nice alternative or add on. 

I will look into it...

![image](https://user-images.githubusercontent.com/19718152/156702981-f6ffb1c6-6fc1-4591-b8ed-c79967d37cf1.png)

just some - sketchy ideas before the day starts

**Reply by tobias-klein** [reply]
@42sol-eu Very nice! When we added icons last year that was a great enhancement to the UI! And there are still quite some functional areas that would benefit from having (better) icons!

As @zhuiks mentioned we currently have two ways to embed icons:
* Using a CSS class ([Font Awesome](https://fontawesome.com/docs/web/setup/get-started) approach)
* Embedding the icon SVG within an element and referencing that element (we use that for the radial verse context menu in fullscreen mode). Examples are [here](https://github.com/ezra-bible-app/ezra-bible-app/blob/master/html/boxes.html#L56) and the icons are referenced [here](https://github.com/ezra-bible-app/ezra-bible-app/blob/master/app/frontend/controllers/wheelnav_controller.js#L127).

PS:
I do not know the remix icon project. But will check it out later.
